diff --git a/media_gallery.pages.inc b/media_gallery.pages.inc
index 2d84ab7..b8dd645 100644
--- a/media_gallery.pages.inc
+++ b/media_gallery.pages.inc
@@ -209,7 +209,7 @@ function media_gallery_reorder_collection($collection, $order) {
  */
 function media_gallery_reorder_gallery($gallery, $order) {
   $media_field = $gallery->media_gallery_media[LANGUAGE_NONE];
-
+  $offset = 0;
   if (!empty($_POST['page'])) {
     $page = array_pop(explode(',', $_POST['page']));
     $num_per_page = $gallery->media_gallery_columns[LANGUAGE_NONE][0]['value'] * $gallery->media_gallery_rows[LANGUAGE_NONE][0]['value'];
